### PR TITLE
[Refactor] Wallet db keys enum + old CWalletKey removal.

### PR DESCRIPTION
**Pull request by furszy**
Making the wallet db nicer to work with, grouping all the keys in a single spot.
Also, removed `CWalletKey` that was never supported in the whole project history.

### CODE REVIEW

File: src/wallet/walletdb.cpp [hunk 1]
```diff
@@ -23,6 +23,44 @@
...
+    const std::string POOL{"pool"};
+    const std::string PURPOSE{"purpose"};
+    const std::string TX{"tx"};
+    const std::string VERSION{"version"};
+    const std::string WATCHS{"watchs"};
+
+    // Sapling
+    const std::string SAP_KEYMETADA{"sapzkeymeta"};
+    const std::string SAPKEY{"sapzkey"};
+    const std::string CRYPTED_SAPKEY{"csapzkey"};
```
Comment (random-zebra) [inline src/wallet/walletdb.cpp:49]: nit: could be good to use the same prefix (`SAP_`) for all related db keys
```suggestion
    const std::string SAP_KEYMETA{"sapzkeymeta"};
    const std::string SAP_KEY{"sapzkey"};
    const std::string SAP_KEY_CRIPTED{"csapzkey"};
```

**Review by random-zebra — COMMENTED** [review]
concept/code ACK. Would be nice to script this change, as checking the lines one by one is a bit tedious.

**Review by random-zebra — DISMISSED** [review]
ACK 995fe6c9645ffd216cd29640b39f2576ba124549

**Comment by random-zebra** [comment]
Can we merge this one?
Quite simple,  and it has been submitted 3 weeks ago.

File: src/wallet/walletdb.cpp [hunk 2]
```diff
@@ -315,35 +351,35 @@ bool CWalletDB::WriteAutoCombineSettings(bool fEnable, CAmount nCombineThreshold
...
+    return batch.Write(std::string(DBKeys::MINVERSION), nVersion);
 }
 
 bool CWalletDB::WriteHDChain(const CHDChain& chain)
 {
     nWalletDBUpdateCounter++;
-    std::string key = std::string("hdchain");
+    std::string key = std::string(DBKeys::HDCHAIN);
     if (chain.chainType == HDChain::ChainCounterType::Sapling)
         key += std::string("_sap");
```
Comment (Fuzzbawls) [inline src/wallet/walletdb.cpp:384]: Nit: can use `SAP_HDCHAIN` here for consistency.
```diff
Index: src/wallet/walletdb.cpp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/wallet/walletdb.cpp b/src/wallet/walletdb.cpp
--- a/src/wallet/walletdb.cpp	(revision ffddf66ac12b3cd6ee2cbc1ea77f46a6d1357785)
+++ b/src/wallet/walletdb.cpp	(date 1617929429351)
@@ -381,7 +381,7 @@
     nWalletDBUpdateCounter++;
     std::string key = std::string(DBKeys::HDCHAIN);
     if (chain.chainType == HDChain::ChainCounterType::Sapling)
-        key += std::string("_sap");
+        key = std::string(DBKeys::SAP_HDCHAIN);
     return batch.Write(key, chain);
 }
 
```

**Review by Fuzzbawls — COMMENTED** [review]
overall good. a minor nit commented inline.

Also, while we're currently never calling `WriteMultiSig`/`EraseMultiSig` from anywhere, could add it to `DBKeys` anyways for completeness.

File: src/wallet/walletdb.cpp (hunk 2 above)
Comment (furszy) [reply src/wallet/walletdb.cpp]: Done, added.

**Comment by furszy** [comment]
Updated per feedback.
...

> Also, while we're currently never calling WriteMultiSig/EraseMultiSig from anywhere, could add it to DBKeys anyways for completeness.

Removed the two `WriteMultiSig and EraseMultiSig` functions.
 Those were just remnants of the previous bad implementation that we missed to remove in #1662. Plus well, the client already has a process to store multisig scripts using `CWallet::AddCScript`. 
So.. good, extra unneeded stuff cleaned.

**Review by Fuzzbawls — APPROVED** [review]
ACK f7e8e1e7d22662d9e0e6a3d2eb6c4328c0963878

**Review by random-zebra — APPROVED** [review]
utACK f7e8e1e7d22662d9e0e6a3d2eb6c4328c0963878 and merging...

**Pull request merged**